### PR TITLE
(O2-932) Cleanup modulefiles

### PR DESCRIPTION
**Pull request by Barthelemy**
- `PACKAGE_VERSION` -> `PACKAGE_REVISION` in guards
- Removal of `$([[ ${ARCHITECTURE:0:3} == osx ]] && echo "prepend-path DYLD_LIBRARY_PATH \$AEGIS_ROOT/lib")`

### CODE REVIEW

**Comment by Barthelemy** [comment]
we discussed with @ktf and decided to wait Monday to merge the huge PR.

**Comment by Barthelemy** [comment]
The code owners are *strongly* encouraged to check and test this PR (and to approve it if it works).

File: ampt.sh [hunk 1]
```diff
@@ -27,10 +27,9 @@ proc ModulesHelp { } {
...
 # Dependencies
-module load BASE/1.0 ${HEPMC_VERSION:+HepMC/$HEPMC_VERSION-$HEPMC_REVISION}
+module load BASE/1.0 ${HEPMC_REVISION:+HepMC/$HEPMC_VERSION-$HEPMC_REVISION}
 # Our environment
 setenv AMPT_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
 prepend-path PATH \$::env(AMPT_ROOT)/bin
 #prepend-path LD_LIBRARY_PATH \$::env(AMPT_ROOT)/lib
-#$([[ ${ARCHITECTURE:0:3} == osx ]] && echo "prepend-path DYLD_LIBRARY_PATH \$::env(AMPT_ROOT)/lib")
-EoF
+#EoF
```
Comment (ktf) [inline ampt.sh]: Are you sure about this?

**Comment by aphecetche** [comment]
Hum, just tested the impact on alo-aliroot and I get non-working module(s) as far as I can tell  

(using aliBuild --defaults next-root6 build alo-aliroot --disable fastjet)

```
[ pr ]~/alice/pr$ module load alo-aliroot
Loading ROOT/v6-18-04-3
  ERROR: Unable to locate a modulefile for 'XRootD/-1'
  ERROR: Requirement XRootD/-1 is not loaded

Loading DPMJET/v3.0.5-alice4-1
  Module ERROR: can't read "::env(DPMJET_ROOT)": no such variable
    In '/Users/laurent/alice/pr/sw/MODULES/osx_x86-64/DPMJET/v3.0.5-alice4-1'
    Please contact <root@localhost>

Loading GEANT3/v2-7-2
  ERROR: Requirement ROOT/v6-18-04-3 is not loaded

Loading GEANT4/v10.4.2-2
  Module ERROR: can't read "::env(GEANT4_ROOT)": no such variable
    In '/Users/laurent/alice/pr/sw/MODULES/osx_x86-64/GEANT4/v10.4.2-2'
    Please contact <root@localhost>

Loading vgm/v4-4-2
  ERROR: Requirement GEANT4/v10.4.2-2 is not loaded

Loading GEANT4_VMC/v3-6-p3-2
  ERROR: Requirement GEANT4/v10.4.2-2 is not loaded

Loading Vc/1.4.1-3
  Module ERROR: can't read "::env(VC_ROOT)": no such variable
    In '/Users/laurent/alice/pr/sw/MODULES/osx_x86-64/Vc/1.4.1-3'
    Please contact <root@localhost>

Loading AliEn-ROOT-Legacy/0.1.3-1
  ERROR: Requirement ROOT/v6-18-04-3 is not loaded

Loading AliRoot/v5-09-51_ROOT6-1
  ERROR: Requirement ROOT/v6-18-04-3 is not loaded

Loading boost/v1.70.0-3
  Module ERROR: can't read "::env(BOOST_ROOT)": no such variable
    In '/Users/laurent/alice/pr/sw/MODULES/osx_x86-64/boost/v1.70.0-3'
    Please contact <root@localhost>

Loading yaml-cpp/yaml-cpp-0.6.2-3
  ERROR: Requirement boost/v1.70.0-3 is not loaded

Loading alo-aliroot/latest-next-root6
  ERROR: Load of requirement XRootD/-1 failed
  ERROR: Load of requirement AliRoot/v5-09-51_ROOT6-1 failed
 ```

While with the master I get the expected : 

```
[ dev ]~/alice/dev$ module load alo-aliroot
Loading alo-aliroot/latest-next-root6
  Loading requirement: BASE/1.0 AliEn-Runtime/v2-19-le-1 GSL/v1.16-2 Python-modules/1.0-2 ROOT/v6-16-00-1 DPMJET/v3.0.5-alice4-1 GEANT3/v2-7-1 GEANT4/v10.4.2-1 vgm/v4-4-1
    GEANT4_VMC/v3-6-p3-1 Vc/1.4.1-2 AliEn-ROOT-Legacy/0.1.1-1 AliRoot/master_ROOT6-1 boost/v1.70.0-2 yaml-cpp/yaml-cpp-0.6.2-2
```

File: ampt.sh (hunk 1 above)
Comment (Barthelemy) [reply ampt.sh]: interesting one, indeed, this is wrong. Fixing.

**Comment by Barthelemy** [comment]
@aphecetche Thank you, I am having a look.

**Comment by Barthelemy** [comment]
@ktf  Let's wait more before merging, we should clearly test it more.

**Comment by Barthelemy** [comment]
@ktf I have debugged a bit the problem detected by Laurent. It seems to come from python.sh which unsets certain variables if Alien-Runtime is installed (line 110) : 
```
# Get OpenSSL and zlib at runtime from AliEn-Runtime if appropriate
[[ $ALIEN_RUNTIME_VERSION ]] && unset OPENSSL_VERSION ZLIB_VERSION
```
As a consequence, the modulefile is wrong because these variables are missing.
Do you understand why they are unset ?

**Comment by Barthelemy** [comment]
After discussing with @ktf I have disabled *_REVISION in addition to *_VERSION in two recipes : root.sh and python.sh. 
A real fix will come in the future.

**Comment by Barthelemy** [comment]
@aphecetche could you try again ?

**Comment by Barthelemy** [comment]
@MichaelLettrich there are no logs from the build system

**Comment by aphecetche** [comment]
@Barthelemy Looks good now : 

```
[ pr ]~/alice/pr$ module load alo-aliroot
[ pr ]~/alice/pr$ module list
Currently Loaded Modulefiles:
 1) alice-pr                  11) GEANT4_VMC/v3-6-p3-2
 2) BASE/1.0                  12) Vc/1.4.1-1
 3) AliEn-Runtime/v2-19-le-2  13) XRootD/v3.3.6-alice2-1
 4) GSL/v1.16-1               14) xalienfs/v1.0.14r1-alice3-2
 5) Python-modules/1.0-2      15) AliEn-ROOT-Legacy/0.1.3-2
 6) ROOT/v6-18-04-2           16) AliRoot/v5-09-51_ROOT6-2
 7) DPMJET/v3.0.5-alice4-1    17) boost/v1.70.0-1
 8) GEANT3/v2-7-2             18) yaml-cpp/yaml-cpp-0.6.2-1
 9) GEANT4/v10.4.2-1          19) alo-aliroot/latest-next-root6
10) vgm/v4-4-2

```

**Comment by Barthelemy** [comment]
@MichaelLettrich @ktf it is now the mac builder that is misbehaving.

**Comment by MichaelLettrich** [comment]
Found the issue.  Looks like something is wrong with the FairMQ repos. all builds are failing because of this.

```bash
2019-12-06@07:18:15:DEBUG:O2Suite:FairMQ:1913: + export GIT_REFERENCE=/build/mirror/fairmq
2019-12-06@07:18:15:DEBUG:O2Suite:FairMQ:1913: + GIT_REFERENCE=/build/mirror/fairmq
2019-12-06@07:18:15:DEBUG:O2Suite:FairMQ:1913: + [[ ! https://github.com/FairRootGroup/FairMQ == '' ]]
2019-12-06@07:18:15:DEBUG:O2Suite:FairMQ:1913: + [[ h != \/ ]]
2019-12-06@07:18:15:DEBUG:O2Suite:FairMQ:1913: + [[ ! -d /build/ci_checks/alidist_checker_macos_o2_ci_1/sw/SOURCES/FairMQ/v1.4.9/v1.4.9 ]]
2019-12-06@07:18:15:DEBUG:O2Suite:FairMQ:1913: + cd /build/ci_checks/alidist_checker_macos_o2_ci_1/sw/SOURCES/FairMQ/v1.4.9/v1.4.9
2019-12-06@07:18:15:DEBUG:O2Suite:FairMQ:1913: + git checkout v1.4.9
2019-12-06@07:18:16:DEBUG:O2Suite:FairMQ:1913: error: pathspec 'v1.4.9' did not match any file(s) known to git
2019-12-06@07:18:16:DEBUG:O2Suite:FairMQ:1913: 
2019-12-06@07:18:16:ERROR:O2Suite:FairMQ:1913: Error while executing /build/ci_checks/alidist_checker_macos_o2_ci_1/sw/SPECS/osx_x86-64/FairMQ/v1.4.9-1/build.sh on `alibuildmac04.cern.ch'.og can be found in /build/ci_checks/alidist_checker_macos_o2_ci_1/sw/BUILD/FairMQ-latest/log
2019-12-06@07:18:16:ERROR:O2Suite:FairMQ:1913: Please upload it to CERNBox/Dropbox if you intend to request support.
2019-12-06@07:18:16:ERROR:O2Suite:FairMQ:1913: Build directory is /build/ci_checks/alidist_checker_macos_o2_ci_1/sw/BUILD/FairMQ-latest/FairMQ.UILD_ERROR=1
+ [[ 1 != '' ]]
+ badge failing
+ [[ 1913 == 1913 ]]

```

**Comment by MichaelLettrich** [comment]
@ktf : I have no idea what to do about this. I cant find any signs of retag or force push at the FairMQ repo. Is it sensible to just nuke sw?

**Comment by ktf** [comment]
IIRC you can just cleanup `fairmq` in the `/build/mirror` folder.

**Comment by ktf** [comment]
(on holidays today, so comments might be delayed).

**Comment by MichaelLettrich** [comment]
git fetch in SOURCES added the new tag sucessfully.  I will watch if new PRs are built propperly and then ask @Barthelemy to bump this for rebuild.

**Comment by Barthelemy** [comment]
@ktf Any idea why travis complains about o2.sh ?

**Comment by ktf** [comment]
@barthelemy i think i fixed the linter.

**Comment by Barthelemy** [comment]
@ktf, ok thank you !

**Comment by Barthelemy** [comment]
@ktf is there also something wrong with the linting of fluka ? Is the colon legal here or not ? 
```
Linting fluka.sh
prepend-path PATH \$FLUKA_ROOT:\$FLUKA_ROOT/flutil
```

**Comment by ktf** [comment]
nope

**Comment by Barthelemy** [comment]
All checks have passed, it was tested with defaults "o2", defaults "o2-dataflow" is being tested, the use case of Laurent is fine.
Could people test it ? 
@ktf do we wait after the holidays to merge it ?

**Comment by Barthelemy** [comment]
I made one more change to preserve the environment variables such as QUALITYCONTROL_ROOT which were removed during the change from `setenv` to `set`. Indeed, I simply don't know who might be using them (and QC was using it per instance). 

@ktf I think that we should wait after Christmas to merge this monster of a PR.

**Comment by Barthelemy** [comment]
I would need some testers for this massive PR. Would the following people agree giving it a try ? I try to cover different parts of the project (FLP, EPN, PDP, detectors). If you can't, no worries, just let me know.

- @sy-c
- @chiarazampolli or @shahor02 
- @teo 
- @ironMann 
- @aferrero2707
- @wiechula 

I am interested to know whether your usual sequence of `aliBuild`, `alienv` and running your software still works. Alternatively you can send me the exact commands you usually run and I will test it. 
Thanks a lot

**Comment by shahor02** [comment]
Hi @Barthelemy 
I'll test (ubuntu 18.04)
Cheers,
 Ruben

**Comment by shahor02** [comment]
@Barthelemy : with this branch the ``aliBuild build O2 --defaults o2`` and running usual workflows works fine.

**Comment by ironMann** [comment]
Hi @Barthelemy ,
On Centos7 I get aliBuild error (current alidist/master works fine, no local development repositories). The command is:
`aliBuild build DataDistribution --debug --jobs 50 --defaults o2-dataflow`

```
DEBUG:DataDistribution:DebugGUI:v0.7: -- Check for working CXX compiler: /home/neskovic/alicetest/sw/slc7_x86-64/GCC-Toolchain/v7.3.0-alice1-1/bin/c++ -- works
DEBUG:DataDistribution:DebugGUI:v0.7: -- Detecting CXX compiler ABI info
DEBUG:DataDistribution:DebugGUI:v0.7: -- Detecting CXX compiler ABI info - done
DEBUG:DataDistribution:DebugGUI:v0.7: -- Detecting CXX compile features
DEBUG:DataDistribution:DebugGUI:v0.7: -- Detecting CXX compile features - done
DEBUG:DataDistribution:DebugGUI:v0.7: -- Could NOT find GLFW (missing: GLFW_DIR)
DEBUG:DataDistribution:DebugGUI:v0.7: CMake Error at DebugGUI/CMakeLists.txt:98 (install):
DEBUG:DataDistribution:DebugGUI:v0.7:   install TARGETS given target "imgui_demo" which does not exist.
DEBUG:DataDistribution:DebugGUI:v0.7:
DEBUG:DataDistribution:DebugGUI:v0.7:
DEBUG:DataDistribution:DebugGUI:v0.7: -- Configuring incomplete, errors occurred!
DEBUG:DataDistribution:DebugGUI:v0.7: See also "/home/neskovic/alicetest/sw/BUILD/b5f0e0c2f7b044b61f0b0b0d33804308935b6721/DebugGUI/CMakeFiles/CMakeOutput.log".
```

**Comment by ktf** [comment]
@Barthelemy i think you need one more merge / rebase

**Comment by Barthelemy** [comment]
@ironMann Thanks a lot for checking it out. I am going to fix it. 
@shahor02 Thank you for the test and the feedback.

**Comment by Barthelemy** [comment]
@ironMann Could you try again ? I merged again with master thanks to @ktf .

**Comment by ironMann** [comment]
@Barthelemy @ktf DebugGUI builds, but I get stuck with the [Python (logs)](https://gist.github.com/ironMann/c96adee1864d3d9cf6f218d5601d61e9):
```
DEBUG:DataDistribution:Python:v0.7: +++ echo python3.6
DEBUG:DataDistribution:Python:v0.7: +++ head -n1
DEBUG:DataDistribution:Python:v0.7: +++ for X in 'python*'
DEBUG:DataDistribution:Python:v0.7: +++ echo python3.6m
DEBUG:DataDistribution:Python:v0.7: +++ for X in 'python*'
DEBUG:DataDistribution:Python:v0.7: +++ echo python3.6m-config
DEBUG:DataDistribution:Python:v0.7: ++ PYTHON_BIN=python3.6
DEBUG:DataDistribution:Python:v0.7: +++ for X in 'pip*'
DEBUG:DataDistribution:Python:v0.7: +++ echo 'pip*'
DEBUG:DataDistribution:Python:v0.7: +++ grep -E '^pip[0-9]+\.[0-9]+$'
DEBUG:DataDistribution:Python:v0.7: +++ head -n1
DEBUG:DataDistribution:Python:v0.7: ++ PIP_BIN=
DEBUG:DataDistribution:Python:v0.7:
ERROR:DataDistribution:Python:v0.7: Error while executing /home/neskovic/alicetest/sw/SPECS/slc7_x86-64/Python/v3.6.8-1/build.sh on `dev01.compeng.uni-frankfurt.de'.
ERROR:DataDistribution:Python:v0.7: Log can be found in /home/neskovic/alicetest/sw/BUILD/Python-latest/log.
ERROR:DataDistribution:Python:v0.7: Please upload it to CERNBox/Dropbox if you intend to request support.
ERROR:DataDistribution:Python:v0.7: Build directory is /home/neskovic/alicetest/sw/BUILD/Python-latest/Python.
```

**Comment by ktf** [comment]
It's listed in the page that says "Prerequisites for Centos7" at the top and the section ends with "You are now ready for installing aliBuild and start building ALICE software". I think we always said for user installations it was mandatory. That said, I agree we should clean that up and not require it. In any case the problem is in downloading LLVM, I am looking into it.

**Comment by Barthelemy** [comment]
@ktf probably wrong thread

**Comment by sy-c** [comment]
Hello @Barthelemy,
I have checked the branch this morning, it's ok to compile and run Readout.

**Comment by Barthelemy** [comment]
@sy-c Thank you!

**Comment by wiechula** [comment]
Hi @Barthelemy ,

I tested on slc7 using alidist@d1fcb21979393c7beab85051cf3b9b79ae59d50e. Compilation works without problems. However, when I try to start e.g. our data monitor I get:
{noformat}
cling::DynamicLibraryManager::loadLibrary(): libO2DebugGUI.so
{noformat}
Ineeded {{DebugGUI}} is not loaded. I assume it is missing in the Module file dependencies somewhere.
{noformat}
grep -r DebugGUI sw/MODULES/*
sw/MODULES/slc7_x86-64/DebugGUI/latest:  puts stderr "ALICE Modulefile for DebugGUI v0.1.0-55c12c2-1"
sw/MODULES/slc7_x86-64/DebugGUI/latest:module-whatis "ALICE Modulefile for DebugGUI v0.1.0-55c12c2-1"
sw/MODULES/slc7_x86-64/DebugGUI/latest:set DEBUGGUI_ROOT $::env(BASEDIR)/DebugGUI/$version
sw/MODULES/slc7_x86-64/DebugGUI/v0.1.0-55c12c2-1:  puts stderr "ALICE Modulefile for DebugGUI v0.1.0-55c12c2-1"
sw/MODULES/slc7_x86-64/DebugGUI/v0.1.0-55c12c2-1:module-whatis "ALICE Modulefile for DebugGUI v0.1.0-55c12c2-1"
sw/MODULES/slc7_x86-64/DebugGUI/v0.1.0-55c12c2-1:set DEBUGGUI_ROOT $::env(BASEDIR)/DebugGUI/$version
sw/MODULES/slc7_x86-64/DebugGUI/latest-dev-o2:  puts stderr "ALICE Modulefile for DebugGUI v0.1.0-55c12c2-1"
sw/MODULES/slc7_x86-64/DebugGUI/latest-dev-o2:module-whatis "ALICE Modulefile for DebugGUI v0.1.0-55c12c2-1"
sw/MODULES/slc7_x86-64/DebugGUI/latest-dev-o2:set DEBUGGUI_ROOT $::env(BASEDIR)/DebugGUI/$version
{noformat}

Cheers,
Jens

**Comment by wiechula** [comment]
Funny, I just tried again in a fresh shell and now everything worked. Perhaps there was some left over env settings. Sorry for the noise.

**Comment by Barthelemy** [comment]
@wiechula Thank you very much for testing it out, I am grateful.

**Comment by ironMann** [comment]
@Barthelemy I've recreated an clean centos7 VM to test this and my workflow seems to be fine. 

With this branch aliBuild ends up building all dependencies from scratch, while for the alidist/master it downloads prebuilt packages (for GCC toolchain etc...). My main dev machine endsup building everything from source, all the time. But that is the story for a different thread...

**Comment by Barthelemy** [comment]
@ironMann Thank you very much for trying it out again. I am happy that it worked. 

@teo @aferrero2707 If you want to give it a try, it is time, otherwise I will ask Giulio to force merge it this week.

**Comment by ktf** [comment]
I went through it again. It looks fine, however in the long term we should make sure we remove any environment variable which is not actually expected by the runtime of a given tool, so that we can pick it from the system if available. In particular all the `_ROOT` variables should be removed. We can do that in a subsequent PR, of course, but we should at least avoid introducing new ones.

**Comment by Barthelemy** [comment]
@ktf ok for a later PR. It will be a tedious one as it is a human process to determine what variables are needed.

**Comment by aferrero2707** [comment]
@Barthelemy I have finally managed to find some time and test this PR. However, on my system things seem to break even before reaching the QC compilation.

This is my build script:
```
#! /bin/bash

export ALIBUILD_WORK_DIR="$HOME/QC-test/alice/sw"
eval "`alienv shell-helper`"

aliBuild init O2@dev --defaults o2-dataflow
aliBuild init QualityControl@attempt-modulefile-cleanup2 --defaults o2-dataflow
aliBuild -j 4 build flpproto --defaults o2-dataflow
```

The compilation stops at DebugGUI, with the following message:
```
-- Could NOT find GLFW (missing: GLFW_DIR)
CMake Error at DebugGUI/CMakeLists.txt:77 (add_dependencies):
  Cannot add target-level dependencies to non-existent target "imgui_demo".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.
```

Any idea?

**Comment by ktf** [comment]
@aferrero2707 can you install `glfw-devel` for the moment and check again? I am fixing the DebugGUI CMakeLists.txt.

**Comment by Barthelemy** [comment]
`sudo yum install -y glfw-devel --enablerepo=epel` should do the trick if you are on CC7.

**Comment by ktf** [comment]
I think I've also a proper fix in `alisw/alidist#1979`.

**Comment by aferrero2707** [comment]
@ktf @Barthelemy installing `glfw-devel` did the trick. Now the compilation of QC stops with the following error:
```
make: Entering directory `/home/aferrero/QC-test/alice/sw/BUILD/3a059e00d10edec1528f335aa731f947976f00ad/qcg/node_modules/@aliceo2/qc/build'
  cc '-DNODE_GYP_MODULE_NAME=nothing' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' -I/home/aferrero/.cache/node-gyp/12.13.0/include/node -I/home/aferrero/.cache/node-gyp/12.13.0/src -I/home/aferrero/.cache/node-gyp/12.13.0/deps/openssl/config -I/home/aferrero/.cache/node-gyp/12.13.0/deps/openssl/openssl/include -I/home/aferrero/.cache/node-gyp/12.13.0/deps/uv/include -I/home/aferrero/.cache/node-gyp/12.13.0/deps/zlib -I/home/aferrero/.cache/node-gyp/12.13.0/deps/v8/include  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -m64 -O3 -fno-omit-frame-pointer  -MMD -MF ./Release/.deps/Release/obj.target/nothing/../../node-addon-api/src/nothing.o.d.raw  -fPIC -O2 -c -o Release/obj.target/nothing/../../node-addon-api/src/nothing.o ../../../node-addon-api/src/nothing.c
  rm -f Release/obj.target/../../node-addon-api/src/nothing.a && ar crs Release/obj.target/../../node-addon-api/src/nothing.a Release/obj.target/nothing/../../node-addon-api/src/nothing.o
  rm -rf "Release/nothing.a" && cp -af "Release/obj.target/../../node-addon-api/src/nothing.a" "Release/nothing.a"
  g++ '-DNODE_GYP_MODULE_NAME=tobject2json' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DNAPI_DISABLE_CPP_EXCEPTIONS' '-DBUILDING_NODE_EXTENSION' -I/home/aferrero/.cache/node-gyp/12.13.0/include/node -I/home/aferrero/.cache/node-gyp/12.13.0/src -I/home/aferrero/.cache/node-gyp/12.13.0/deps/openssl/config -I/home/aferrero/.cache/node-gyp/12.13.0/deps/openssl/openssl/include -I/home/aferrero/.cache/node-gyp/12.13.0/deps/uv/include -I/home/aferrero/.cache/node-gyp/12.13.0/deps/zlib -I/home/aferrero/.cache/node-gyp/12.13.0/deps/v8/include -I/home/aferrero/QC-test/alice/sw/BUILD/3a059e00d10edec1528f335aa731f947976f00ad/qcg/node_modules/node-addon-api -I/home/aferrero/QC-test/alice/sw/slc7_x86-64/QualityControl/master-1/include -I/home/aferrero/QC-test/alice/sw/slc7_x86-64/ROOT/v6-18-04-1/include -I/home/aferrero/QC-test/alice/sw/slc7_x86-64/Configuration/v2.2.4-1/include -I/home/aferrero/QC-test/alice/sw/slc7_x86-64/Common-O2/v1.4.6-1/include -I/home/aferrero/QC-test/alice/sw/slc7_x86-64/boost/v1.70.0-1/include  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -m64 -O3 -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++1y -std=c++1z -frtti -MMD -MF ./Release/.deps/Release/obj.target/tobject2json/tobject2json.o.d.raw  -fPIC -O2 -std=c++17 -c -o Release/obj.target/tobject2json/tobject2json.o ../tobject2json.cc
../tobject2json.cc: In member function 'virtual void TObjectAsyncWorker::Execute()':
../tobject2json.cc:23:31: error: 'class o2::quality_control::repository::DatabaseInterface' has no member named 'retrieveJson'; did you mean 'retrieveMOJson'?
     output = BackendInstance->retrieveJson(path.substr(0, slashIndex), path.substr(slashIndex + 1));
                               ^~~~~~~~~~~~
                               retrieveMOJson
make: *** [Release/obj.target/tobject2json/tobject2json.o] Error 1
make: Leaving directory `/home/aferrero/QC-test/alice/sw/BUILD/3a059e00d10edec1528f335aa731f947976f00ad/qcg/node_modules/@aliceo2/qc/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/aferrero/QC-test/alice/sw/slc7_x86-64/node/v12.13.0-1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:210:5)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Linux 3.10.0-1062.9.1.el7.x86_64
gyp ERR! command "/home/aferrero/QC-test/alice/sw/slc7_x86-64/node/v12.13.0-1/bin/node" "/home/aferrero/QC-test/alice/sw/slc7_x86-64/node/v12.13.0-1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--jobs" "max"
gyp ERR! cwd /home/aferrero/QC-test/alice/sw/BUILD/3a059e00d10edec1528f335aa731f947976f00ad/qcg/node_modules/@aliceo2/qc
gyp ERR! node -v v12.13.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 
```

Let me know if you need the whole build log...

**Comment by ktf** [comment]
@awegrzyn ideas?

**Comment by Barthelemy** [comment]
the PR merged yesterday (qc and qcg bump in alidist) should fix this. Could you make sure that you have the latest alidist ?

**Comment by aferrero2707** [comment]
@Barthelemy should I take the official alidist or your `attempt-modulefile-cleanup2` branch?

**Comment by awegrzyn** [comment]
There was a change in QC that broke QCG, so make sure you use qcg v1.6.9 and QC 0.20.x

**Comment by aferrero2707** [comment]
@awegrzyn @Barthelemy in the current `attempt-modulefile-cleanup2` branch, qcg is still at version 1.6.8:
```
package: qcg
version: v1.6.8
tag: "@aliceo2/qc@1.6.8"
```

**Comment by aferrero2707** [comment]
@Barthelemy the latest commit allowed me to successfully complete the compilation of QC. Here is the list of enabled modules on my system:
```
Currently Loaded Modulefiles:
  1) BASE/1.0                       15) zlib/v1.2.8-1                  29) Common-O2/v1.4.6-1
  2) GCC-Toolchain/v7.3.0-alice1-1  16) OpenSSL/v1.0.2o-1              30) ms_gsl/1-1
  3) FairLogger/v1.5.0-1            17) libxml2/v2.9.3-1               31) O2/0_O2_DATAFLOW-1
  4) libpng/v1.6.34-1               18) GSL/v1.16-1                    32) InfoLogger/v1.3.7-1
  5) libffi/v3.2.1-1                19) Python-modules/1.0-1           33) PDA/12.0.0-1
  6) FreeType/v2.6-1                20) arrow/v0.14.1-1                34) ReadoutCard/v0.15.4-1
  7) Python/v3.6.8-1                21) lzma/v5.2.3-1                  35) lz4/v1.8.3-1
  8) boost/v1.70.0-1                22) ROOT/v6-18-04-1                36) c-ares/v1.15.0-1
  9) ZeroMQ/v4.1.5-1                23) FairRoot/6af7e84858-1          37) grpc/v1.19.1-1
 10) ofi/v1.7.1-1                   24) Vc/1.4.1-1                     38) Control-OCCPlugin/v0.11.1-1
 11) asiofi/v0.3.1-1                25) Monitoring/v2.6.3-1            39) Readout/v1.2.5-1
 12) DDS/2.4-1                      26) Ppconsul/0.1.0-1               40) QualityControl/master-1
 13) FairMQ/v1.4.9-1                27) Configuration/v2.2.4-1         41) flpproto/latest-o2-dataflow
 14) protobuf/v3.7.1-1              28) libInfoLogger/v1.3.7-1
Use alienv list to list loaded modules. Use exit to exit this environment.
```

Is there anything else you need me to check?

**Comment by Barthelemy** [comment]
@aferrero2707 no it is ok like that, thank you very much !

**Comment by victor-gonzalez** [comment]
The modules load in Ubuntu 18.04 has been broken.
```
victor@ganimedes:~$ alienv load AliPhysics/latest-alinextroot6-user-next-root6
Use alienv list to list loaded modules.
AliPhysics/latest-alinextroot6-user-next-root6(34):ERROR:102: Tcl command execution failed: prepend-path PATH $ALICE_PHYSICS/bin
victor@ganimedes:~$ alienv list
No Modulefiles Currently Loaded.
```
I saw that in ``aliphysics.sh``, ``::env()`` has been removed from
```
prepend-path PATH \$::env(ALICE_PHYSICS)/bin
```
and other similar, which were present before.
It seems Ubuntu does not like such a removal.
I changed back and producing again to check that was the problem.
Is it feasible to keep ``::env()``? is there any other workaround?

Thanks in advance!

**Comment by ktf** [comment]
Can you check if #2016 fixed the issue already?

**Comment by victor-gonzalez** [comment]
It will take a while.
I will come back with the result.

**Comment by Barthelemy** [comment]
@ktf you are fast ! 
I had the time to spawn my Ubuntu VM while you already proposed a solution :)

**Comment by victor-gonzalez** [comment]
Both approaches worked, mine and #2016, so, I stick to the official one.
Thanks!

**Pull request merged**